### DISCUSSION TITLE
sending log file away from pico w

### DISCUSSION THREAD

**Discussion by fishjoe** [micropython/micropython#10436] [RP2040 / Pico]
Hi there,

I have set-up a device run automatically without connecting to computer. It will make a log file named log.json. 

What is the best way you guys think to send the data from this json file away and keep a record at somewhere I can easily access? 

i.e (not necessarily practical, just an example) save it on iCloud or dropbox?

**Comment by GM-Script-Writer-62850** [comment]
You could just send data you want logged on the fly to a server and not log it on the pico, i assume you are using the onboard flash storage for your log file, flash nand does not have a inf write limit so eventually this will wear it out, you could wire a sd card to the pico

i made a rather complex sending method so if my main loop makes 5 log entries in a single loop pass I only need 1 TCP connection to send all 5

* example code: https://forums.raspberrypi.com/viewtopic.php?t=341819 but instead of using print you could use urequests
  * real world usage: https://github.com/GM-Script-Writer-62850/PICO_W_Thermostat/blob/main/PICO/main.py#L142
    * custom modified urequest lib in use in this project (none of the mods apply to this part of how it is being used)
    * The PHP used to process the log data the pico is sending is here: https://github.com/GM-Script-Writer-62850/PICO_W_Thermostat/blob/main/server/index.php#L58-L87
      * This one script handles multiple actions, log view, log saving (as well deleting old log data), sending the PICO saved config data, and telling the PICO when DST changes happen for the given year

if you use `urequests` (`print` was used in the example) to send the data to a server (eg: raspberry pi  or a x86 based computer) on your network (cause we want logs to work if the internet is down) you can handle it anyway you want and not have to worry about getting to a sd card, wearing out your flash storage, or a lack or storage space
